assets/js/phoenix_live_view/view.js

```diff
@@ -970,9 +970,9 @@ export default class View {
             finish(null)
           }
         },
-        error: (reason) => reject({error: reason}),
+        error: (reason) => reject(new Error(`failed with reason: ${reason}`)),
         timeout: () => {
-          reject({timeout: true})
+          reject(new Error("timeout"))
```

Comment:
If one day it becomes important to differentiate errors downstream, we could either have custom error classes or error codes.

For now, I think we're using `new Error(...)` throughout the code base for simplicity.